src/workflows/multiomics/process_batches/main.nf

```diff
@@ -172,6 +177,24 @@ workflow run_wf {
                     "var_pca_feature_selection": state.highly_variable_features_var_output, // run PCA on highly variable genes only
                     "pca_overwrite": state.pca_overwrite,
                   ],
+                "dimensionality_reduction_scaling_rna":
+                  [
+                    "id": id,
+                    "input": state.input,
+                    "layer": state.rna_scaling_output_layer,
+                    "modality": "rna",
+                    "var_pca_feature_selection": state.highly_variable_features_var_output, // run PCA on highly variable genes only
+                    "pca_overwrite": state.pca_overwrite,
+                    // extra scaling args
+                    "obsm_pca": state.rna_scaling_pca_obsm_output,
+                    "pca_loadings_varm_output": state.rna_scaling_pca_loadings_varm_output,
+                    "pca_variance_uns_output": state.rna_scaling_pca_variance_uns_output,
+                    "pca_overwrite": state.pca_overwrite,
+                    "obsm_umap": state.rna_scaling_umap_obsm_output,
+                    "uns_neighbors": "neighbors_scaled",
+                    "obsp_neighbor_connectivities": "connectivities_scaled",
+                    "obsp_neighbor_distances": "distances_scaled",
+                  ],
```

Comment:
see previous comment. I would remove duplicated code and only add the scaling args if `state.enable_rna_scaling` without needing to duplicate dependencies

Reply:
See above